gensim/test/test_fasttext.py

```diff
@@ -497,12 +497,12 @@ def test_sg_hs_training(self):
             overlap_count, 2,
             "only %i overlap in expected %s & actual %s" % (overlap_count, expected_sims_words, sims_gensim_words))
 
-    def test_sg_hs_training_fromfile(self):
+    def test_sg_hs_training_fromfile(self, shrink_windows=True):
```

Comment:
https://docs.pytest.org/en/6.2.x/example/parametrize.html

```suggestion
    @pytest.mark.parametrize('shrink_windows', [True, False])
    def test_sg_hs_training_fromfile(self, shrink_windows):
```

Reply:
I pushed two commits making use of that feature. The first one only applied to the FastText tests, as they were the one you attached this review to. The second one applies to the other two test sets (wor2vec and doc2vec); if you feel like the latter should not have been changed, we can discard this commit.

Reply:
Update: actually, `pytest.mark.parametrize` does not work as expected on methods (although it does on functions). I will try to find a way to make it work, otherwise I will undo the last two commits.

Reply:
Oh, in fact it probably will not work due to the test classes being `unittest.TestCase` subclasses.

https://docs.pytest.org/en/6.2.x/unittest.html
> The following pytest features do not work [in unittest.TestCase subclasses], and probably never will due to different design philosophies: Fixtures (except for autouse fixtures, see below); Parametrization; Custom hooks;

Reply:
I am revoking the last two commits as a consequence.

Reply:
Thanks for pointing that out. I didn't know pytest parameterization is incompatible with unittest.TestCase.

Luckily, in this particular case, the unittest.TestCase class isn't doing anything. I'll get parameterization working in a separate commit.

Reply:
Well, we both learned something :)